core/Pos/Core/Genesis/Types.hs

```diff
@@ -57,20 +48,44 @@ getTotalStake ExponentialStakes = mkCoin . sum $
     let g 0 = []
         g n = n : g (n `div` 2)
     in g 5000
-getTotalStake (ExplicitStakes balances) = unsafeIntegerToCoin $
-    sumCoins $ fst <$> balances
-getTotalStake (CombinedStakes st1 st2) =
-    getTotalStake st1 `unsafeAddCoin` getTotalStake st2
+getTotalStake (CustomStakes balances) =
+    unsafeIntegerToCoin $ sumCoins balances
 
--- | Hardcoded genesis data
-data GenesisCoreData = GenesisCoreData
-    { gcdAddresses         :: [Address]
-    , gcdDistribution      :: StakeDistribution
-    , gcdBootstrapBalances :: !(HashMap StakeholderId Coin)
+-- | List of distributions accompained by related addresses set (what
+-- to distribute and how).
+type AddrDistribution = [(HashSet Address, StakeDistribution)]
```

Comment:
I don't get this alias. Suppose I have `(HS.fromList [a1, a2], CustomStakes [mkCoin 1, mkCoin 2])`. Who will have 1 and who will have 2?